pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -430,8 +501,15 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[network.
 	}
 
 	localityLbEndpoints := make([]*endpoint.LocalityLbEndpoints, 0, len(lbEndpoints))
-
-	for locality, eps := range lbEndpoints {
+	locs := make([]string, 0, len(lbEndpoints))
```

Comment:
```suggestion
	locs := lbEndpoints
```
is there a semantic difference?

Reply:
Yes, one creates an empty list and one makes an alias to a map. We need the list to retain sorted order for deterministic XDS output